devito/types/dense.py

```diff
@@ -1311,3 +1311,35 @@ def parent(self):
         return self._parent
 
     _pickle_kwargs = Function._pickle_kwargs + ['parent']
+
+
+class SeparableFunction(Differentiable):
+    """
+    A function separable in its dimensions, ie f(x,y,z) = f(x)*f(y)*f(z)
+    """
+    def __new__(self, *args, **kwargs):
+        name = kwargs.get('name')
+        grid = kwargs.get('grid')
+        new_obj = np.prod([Function(name=name+'_'+d.name, dimensions=(d,),
+                                    shape=(d.shape,), space_order=kwargs.get('space_order', 1))
+                           for d in grid.dimensions])
+        return new_obj
+        
+class SeparableTimeFunction(Differentiable):
+    """
+    A function separable in its dimensions, ie f(x,y,z) = f(x)*f(y)*f(z)
```

Comment:
might take `t` into account at some point in the description maybe?

Reply:
yes